enzyme/test/MLIR/llvm.mlir

```diff
@@ -0,0 +1,31 @@
+// RUN: %eopt --enzyme %s | FileCheck %s
+
+module {
+  func.func @square(%x : f64) -> f64 {
+    %c1_i64 = arith.constant 1 : i64
+    %tmp = llvm.alloca %c1_i64 x i32 : (i64) -> !llvm.ptr<f64>
+    %y = arith.mulf %x, %x : f64
+    llvm.store %y, %tmp : !llvm.ptr<f64>
+    %r = llvm.load %tmp : !llvm.ptr<f64>
+    return %r : f64
+  }
+  func.func @dsq(%x : f64, %dx : f64) -> f64 {
+    %r = enzyme.fwddiff @square(%x, %dx) { activity=[#enzyme<activity enzyme_dup>] } : (f64, f64) -> (f64)
+    return %r : f64
+  }
+}
+
+// CHECK:   func.func private @fwddiffesquare(%arg0: f64, %arg1: f64) -> f64 {
```

Comment:
Nit: here and below, could we avoid matching SSA names? We have https://github.com/llvm/llvm-project/blob/main/mlir/utils/generate-test-checks.py that replaces them with FileCheck names automatically.